iothub/device/tests/NumericHelpersTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class NumericHelpersTests
+    {
+        [TestMethod]
+        public void CanConvertNumericTypes()
+        {
+            TestNumericConversion<float>(1.001d, true, 1.001f);
```

Comment:
MS test doesn't support parameterized tests with generic types, so implementing it this way